main.tf

```diff
@@ -38,24 +45,16 @@ resource "google_compute_shared_vpc_host_project" "shared_vpc_host" {
 	Subnet configuration
  *****************************************/
 resource "google_compute_subnetwork" "subnetwork" {
-  count = length(var.subnets)
-
-  name                     = var.subnets[count.index]["subnet_name"]
-  ip_cidr_range            = var.subnets[count.index]["subnet_ip"]
-  region                   = var.subnets[count.index]["subnet_region"]
-  private_ip_google_access = lookup(var.subnets[count.index], "subnet_private_access", "false")
-  enable_flow_logs         = lookup(var.subnets[count.index], "subnet_flow_logs", "false")
+  for_each                 = local.subnets
+  name                     = each.value.subnet_name
+  ip_cidr_range            = each.value.subnet_ip
+  region                   = each.value.subnet_region
+  private_ip_google_access = lookup(each.value, "subnet_private_access", "false")
+  enable_flow_logs         = lookup(each.value, "subnet_flow_logs", "false")
```

Comment:
This is outside of the scope of this review so feel free to ignore, but I would very much like doing away with all the redundant prefixes in attribute names. Eg `subnet_name` should imho be just `name`, it's evident it refers to the subnet. Same for outputs like `network_name`, which should just be `name` as the module deals with a VPC and it's obvious what the output refers to.

Reply:
I figured I'd pick up work on #69 and since that work is already changing around the inputs I could remove the prefixes in the keys in that PR as well. Removing the `subnet_` prefix makes sense to me as well.

Reply:
Thanks! :)